state_machine/applications/flight/lib/mathutils.py

```diff
@@ -0,0 +1,40 @@
+from numpy import identity
+
+
+try:
+    from ulab.numpy import array, block, ndarray
+except Exception:
```

Comment:
Out of curiosity, there a reason why we're using numpy as a backup import? I was under the impression that our microcontroller couldn't support it. On the other hand, if we're running this locally (not on the board), why are we opting to import ulab first of just importing numpy?

Reply:
Just to preface, I don't like the way I am doing this. I just haven't figured out a better way to do this.

So ulab for some reason isn't easily available as a pure python package (I might try to fix this), so I can't just import it when running the emulated sat. I'm importing ulab first to prevent unnecessary import failures on the microcontroller because performance matters there.

Reply:
Gotcha, makes sense!